File: frontend/src/services/candidateService.js
import { apiRequest } from "./api";

export const createCandidate = async (candidateData) => {
  return apiRequest("/api/candidates", {
    method: "POST",
    body: JSON.stringify(candidateData),
  });
};

export const getAllCandidates = async (
  pageNo = 0,
  pageSize = 10,
  sortBy = "id",
  sortDir = "asc",
) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
    sortBy,
    sortDir,
  });

  return apiRequest(`/api/candidates?${params}`);
};

export const getCandidateById = async (id) => {
  return apiRequest(`/api/candidates/${id}`);
};

export const updateCandidate = async (id, candidateData) => {
  return apiRequest(`/api/candidates/${id}`, {
    method: "PUT",
    body: JSON.stringify(candidateData),
  });
};

export const deleteCandidate = async (id) => {
  return apiRequest(`/api/candidates/${id}`, {
    method: "DELETE",
  });
};

export const searchCandidates = async (
  candidateName,
  pageNo = 0,
  pageSize = 10,
) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
  });

  return apiRequest(
    `/api/candidates/candidateName/${encodeURIComponent(candidateName)}?${params}`,
  );
};

export const getCandidatesByStatus = async (
  status,
  pageNo = 0,
  pageSize = 10,
) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
  });

  return apiRequest(
    `/api/candidates/status/${encodeURIComponent(status)}?${params}`,
  );
};

File: backend/src/main/java/com/amritan/backend/service/CandidateService.java
package com.amritan.backend.service;


import com.amritan.backend.dto.CandidateDto;
import com.amritan.backend.dto.PageResponse;
import com.amritan.backend.enums.CandidateStatus;

public interface CandidateService {

	CandidateDto createCandidate(CandidateDto candidateDTO);
	
	PageResponse<CandidateDto> getAllCandidates(
	        int pageNo,
	        int pageSize,
	        String sortBy,
	        String sortDir
	);

    CandidateDto getCandidateById(Long id);

    CandidateDto updateCandidate(Long id, CandidateDto candidateDTO);

    void deleteCandidate(Long id);
    
    PageResponse<CandidateDto> searchCandidates(
            String keyword,
            int pageNo,
            int pageSize
    );
    
    PageResponse<CandidateDto> getCandidatesByStatus(
            CandidateStatus status,
            int pageNo,
            int pageSize
    );

}
